packages/block-editor/src/components/rich-text/index.js

```diff
@@ -286,7 +292,15 @@ class RichTextWrapper extends Component {
 	}
 
 	getAllowedFormats() {
-		const { allowedFormats, formattingControls } = this.props;
+		const {
+			allowedFormats,
+			formattingControls,
+			__unstableDisableFormats,
```

Comment:
Instead of having these disbleFormats props which doesn't make sense for a RichText component, ideally RichText should be a special version of Text. how feasable is that?

Reply:
1. Why does it not make sense for `RichText`? I think it's fine.
2. It's harder right now to make `RichText` a wrapper around `Text`, but yes, in the future this could be done. To keep things simple with a minimum amount of code changes, I chose to use `RichText` internally in `Text`, but that can be refactored later.

Reply:
I guess we could just make it a stable prop and omit `__unstable`. Maybe I'm being too prudent in introducing new props.

Reply:
I agree with @ellatrix.
This is a great starting point: it makes things very straightforward, and doesn't prevent an eventual `Text`/`RichText` "switcheroo".

Reply:
In my mind ideally disableFormats could be a valid prod for a low-level Text component that is used by higher-level PlainText and  RichText components. So I think it doesn't make sense for RichText because RichText by definition is for Rich Text which means "formats".

I get that the refactor can be achieved later and is probably not that easy so unstable is better.

Reply:
@youknowriad Oh, I'm sorry, I confused `disableFormats` and `disableLinebreaks`. I thought you were talking about the line breaks. Sorry about that. You're right, the prop doesn't make sense on `RichText` at all, it's just a stepping stone for now.